lib/senders/email.js

```diff
@@ -732,22 +746,30 @@ module.exports = function (log) {
       headers[X_SES_MESSAGE_TAGS] = sesMessageTagsHeaderValue(templateName)
     }
 
-    return this.send(Object.assign({}, message, {
-      headers,
-      subject: gettext('New sign-in to Firefox'),
-      template: templateName,
-      templateValues: {
-        device: this._formatUserAgentInfo(message),
-        ip: message.ip,
-        location: this._constructLocationString(message),
-        passwordChangeLink: links.passwordChangeLink,
-        passwordChangeLinkAttributes: links.passwordChangeLinkAttributes,
-        privacyUrl: links.privacyUrl,
-        supportLinkAttributes: links.supportLinkAttributes,
-        supportUrl: links.supportUrl,
-        timestamp: this._constructLocalTimeString(message.timeZone, message.acceptLanguage)
-      }
-    }))
+    return clientName.fetchClientData(message.service).then((clientData) => {
+      const serviceHeader = translator.format(translator.gettext('New sign-in to %(serviceName)s'), {
```

Comment:
This value goes straight into an email header, I wonder if it's worth doing some sanity-checking and/or escaping on `serviceName` to double-check that it doesn't contain anything dangerous.  In theory it's data we control, but it's adjacent to attacker-provided data (e.g. if a malicious site managers to register a name with embedded newlines, or an attacker manages to trick us into fetching the JSON blob from a malicious URL via carefully crafted `clientId`) so some extra defense might be prudent.